app/models/miq_worker/container_common.rb

```diff
@@ -108,7 +108,7 @@ def deployment_prefix
     def worker_deployment_name
       @worker_deployment_name ||= begin
         deployment_name = abbreviated_class_name.dup.chomp("Worker").sub("Manager", "").sub(/^Miq/, "")
-        deployment_name << "-#{Array(ems_id).map { |id| ApplicationRecord.split_id(id).last }.join("-")}" if respond_to?(:ems_id)
+        deployment_name << "-#{ApplicationRecord.split_id(ems_id).last}" if respond_to?(:ems_id) && ems_id.kind_of?(Fixnum)
```

Comment:
@jrafanie As discussed, let's change the kind_of?(FixNum) to a nil/presence check.  Otherwise this LGTM.

Reply:
ok, done.  PTAL